### PR TITLE
Wrap Backprop* functions to Python.

### PR DESCRIPTION
**Pull request by csukuangfj**
~Not tested yet.~

~Working in progress.~

### CODE REVIEW

**Comment by csukuangfj** [comment]
Ready for review.

File: k2/python/k2/fsa.py [hunk 1]
```diff
@@ -644,33 +656,35 @@ def get_arc_post(self, use_double_scores: bool,
...
           and non-positive elements.
         '''
         # We don't cache this!  User should store it if needed more than once,
         # to avoid duplicate code in backprop.  We may be able to partially fix
         # this at some point with a weak dictionary.
-        forward_scores = self._get_forward_scores(use_double_scores,
-                                                  log_semiring)
-        backward_scores = self._get_backward_scores(use_double_scores,
-                                                    log_semiring)
+        forward_scores = self.get_forward_scores(use_double_scores,
```
Comment (danpovey) [inline k2/python/k2/fsa.py:679]: An issue for later is to try to use a weak dictionary to introduce a caching mechanism for the results of  get_forward_scores and get_backward_scores and get_arc_post (and similar differentiable things), so we don't end up doing the backprop multiple times in cases where, say, the forward and backward scores are used multiple times (which will be common).

**Review by danpovey — COMMENTED** [review]
LGTM!!

**Review by danpovey — APPROVED** [review]

File: k2/python/k2/fsa.py (hunk 1 above)
Comment (csukuangfj) [reply k2/python/k2/fsa.py:679]: Created an issue for it https://github.com/k2-fsa/k2/issues/601

**Comment by csukuangfj** [comment]
Merging.

**Pull request merged**